contracts/05-tamagotchi-auto/io/src/lib.rs

```diff
@@ -1,40 +1,317 @@
 #![no_std]
 
-use gmeta::Metadata;
-use gstd::prelude::*;
+use gmeta::{Metadata, In, Out, InOut};
+use gstd::{prelude::*, msg, ActorId, exec, ReservationId};
+
+use sharded_fungible_token_io::{ FTokenAction, FTokenEvent, LogicAction };
+use store_io::{ StoreAction, StoreEvent };
+
+pub type TransactionId = u64;
+pub type AttributeId = u32;
+
+pub const DELAY_OF_ONE_MINUTE: u32 = 20;
+pub const HUNGER_PER_BLOCK: u64 = 1;
+pub const ENERGY_PER_BLOCK: u64 = 2;
+pub const BOREDOM_PER_BLOCK: u64 = 2;
+pub const FILL_PER_SLEEP: u64 = 1000;
+pub const FILL_PER_FEED: u64 = 1000;
+pub const FILL_PER_ENTERTAINMENT: u64 = 1000;
+
+pub struct GasReservationHandlers {
+    pub contract_send_a_delayed_message: bool,
+    pub can_send_delayed_message: bool
+}
 
 #[derive(Default, Encode, Decode, TypeInfo)]
 #[codec(crate = gstd::codec)]
 #[scale_info(crate = gstd::scale_info)]
 pub struct Tamagotchi {
     // TODO: 0️⃣ Copy fields from previous lesson and push changes to the master branch
+    pub name: String,
+    pub date_of_birth: u64,
+    pub owner: ActorId,
+    pub fed: u64,
+    pub fed_block: u64,
+    pub entertained: u64,
+    pub entertained_block: u64,
+    pub rested: u64,
+    pub rested_block: u64,
+    pub approved_account: Option<ActorId>, 
+    pub ft_contract_id: ActorId,
+    pub transaction_id: TransactionId,
+    pub approve_transaction: Option<(TransactionId, ActorId, u128)>,
```

Comment:
Please don't include changes from previous homework to the PR. Merge them to the master branch as it is mentioned in the comment.